### ISSUE TITLE
[BUG] - Tree shaking doesn't work properly

### ISSUE THREAD

**Issue by HugeLetters** [Maiquu/nuxt-quasar#68]
Thanks in advance for the help!

### Checks

- [X] I have searched the existing issues for a similar bug and it does not exist!

### Environment

- Operating System: Windows_NT
- Node Version:     v20.1.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  pnpm@8.6.7
- Builder:          vite
- User Config:      modules
- Runtime Modules:  nuxt-quasar-ui@2.0.3
- Build Modules:  

### Quasar version

2.12.2

### Browsers

Google Chrome

### Description

Just by providing `"nuxt-quasar-ui"` plugin in `nuxt.config.ts` all of the quasar library gets included in the bundle and gets shipped to the client even if you don't use anything from Quasar library in your code.

Screenshot from `nuxt analyze`
![image](https://github.com/Maiquu/nuxt-quasar/assets/119697239/580ee376-5e83-4b0a-8bb4-66a6a9815e89)

Network tab with module included in config
![image](https://github.com/Maiquu/nuxt-quasar/assets/119697239/d60f377a-53c1-48f8-8d7f-3428de4889ab)

And without
![image](https://github.com/Maiquu/nuxt-quasar/assets/119697239/a7870629-ccc4-4e7c-a8fd-fb918d1548ba)

### Reproduction steps

```bash
1. clone repo with  a starter nuxt template - `git clone git@github.com:HugeLetters/quasar-bundle-test.git`
2. run `pnpm nuxt build`
3. look at the bundle size
4. remove `nuxt-quasar-ui` from nuxt config
5. run 2 & 3 
6. bundle size is reduced
```

### Screenshots

```bash
```

### Logs

```bash
With quasar module
ℹ .nuxt/dist/client/manifest.json                                              3.02 kB │ gzip:  0.50 kB                                                   17:48:08
ℹ .nuxt/dist/client/_nuxt/flUhRq6tzZclQEJ-Vdg-IuiaDsNcIhQ8tQ.4a4dbc62.woff2  128.62 kB                                                                    17:48:08   
ℹ .nuxt/dist/client/_nuxt/flUhRq6tzZclQEJ-Vdg-IuiaDsNa.fd84f88b.woff         164.91 kB                                                                    17:48:08   
ℹ .nuxt/dist/client/_nuxt/error-500.7376934c.css                               1.95 kB │ gzip:  0.74 kB                                                   17:48:08   
ℹ .nuxt/dist/client/_nuxt/error-404.1dfe7e06.css                               3.63 kB │ gzip:  1.12 kB                                                   17:48:08   
ℹ .nuxt/dist/client/_nuxt/index.3dabcfbb.js                                    0.17 kB │ gzip:  0.17 kB                                                   17:48:08   
ℹ .nuxt/dist/client/_nuxt/error-500.ad0bd975.js                                1.88 kB │ gzip:  0.99 kB                                                   17:48:08   
ℹ .nuxt/dist/client/_nuxt/error-404.634a166c.js                                6.53 kB │ gzip:  2.91 kB                                                   17:48:08
ℹ .nuxt/dist/client/_nuxt/entry.20c4bcb5.js                                  243.90 kB │ gzip: 87.51 kB                                                   17:48:08   
ℹ ✓ built in 10.25s             

Without quasar module
ℹ .nuxt/dist/client/manifest.json                   2.31 kB │ gzip:  0.37 kB                                                                              17:45:16   
ℹ .nuxt/dist/client/_nuxt/error-500.7376934c.css    1.95 kB │ gzip:  0.74 kB                                                                              17:45:16
ℹ .nuxt/dist/client/_nuxt/error-404.1dfe7e06.css    3.63 kB │ gzip:  1.12 kB                                                                              17:45:16   
ℹ .nuxt/dist/client/_nuxt/index.04e958e3.js         0.17 kB │ gzip:  0.17 kB                                                                              17:45:16   
ℹ .nuxt/dist/client/_nuxt/error-500.29d9baa0.js     1.88 kB │ gzip:  0.99 kB                                                                              17:45:16   
ℹ .nuxt/dist/client/_nuxt/error-404.28eff166.js     6.53 kB │ gzip:  2.90 kB                                                                              17:45:16   
ℹ .nuxt/dist/client/_nuxt/entry.6b65243f.js       139.17 kB │ gzip: 52.65 kB                                                                              17:45:16   
ℹ ✓ built in 4.48s
```

**Comment by JasonLandbridge** [comment]
Same issue here, I was just about to ask this.

**Comment by JasonLandbridge** [comment]
Just a tip: The [Quasar-Vite plugin](https://quasar.dev/start/vite-plugin/) talks about tree-shaking so maybe some of that can be incorporated as Nuxt runs on Vite as well

**Comment by Maiquu** [comment]
For some reason I had only marked composables, components and directives as side-effect free via this [plugin](https://github.com/Maiquu/nuxt-quasar/blob/v2.0.3/src/plugins/resolveSideEffects.ts). But now that I look at it, this plugin seems unnecessary since this module already takes over the resolve process of `quasar` import via this [plugin](https://github.com/Maiquu/nuxt-quasar/blob/v2.0.3/src/plugins/virtual/entry.ts).

So instead I marked the entry directly as side-effects free like [this](https://github.com/Maiquu/nuxt-quasar/commit/bd902970cd30574eec4b76a7bfd72fe06acb7ef7) which solves the issue with tree-shaking entirely. But then again I need to make sure that if it is safe to do so since quasar devs [refused](https://github.com/quasarframework/quasar/issues/9157#issuecomment-1092118361) to set `sideEffects: false` in `package.json` level due to possibility of breaking existing setups. But then again, it might not apply to this module.

**Comment by Maiquu** [comment]
Closed by ([bd90297](https://github.com/Maiquu/nuxt-quasar/commit/bd90297))

**Comment by Maiquu** [comment]
Note that there will still be imports from `quasar` due to a runtime plugin being injected by the module.
